utils/__tests__/getDataUtils.spec.ts

```diff
@@ -0,0 +1,158 @@
+      expect(parsedCategory.nameUk).toEqual(mockCategoryData.name.source);
+      expect(parsedCategory.translations[0].getId()).toEqual(mockPhraseData.id);
+    });
+  });
+
+  describe('getCategories', () => {
+    it('should return an array of Category objects', () => {
+      const categories = getCategories(mockDictionaryData);
+      expect(categories.length).toEqual(1);
+      expect(categories[0].id).toEqual(mockCategoryData.id);
+    });
+  });
+
+  describe('getAllPhrases', () => {
+    it('should return an array of Phrase objects', () => {
+      const phrases = getAllPhrases(mockDictionaryData);
+      expect(phrases.length).toEqual(1);
+      expect(phrases[0].getId()).toEqual(mockPhraseData.id);
+    });
+  });
+
+  describe('getKidsCategory', () => {
+    it('should return the kids category if present', () => {
+      const kidsCategoryData = { ...mockCategoryData, id: 'recSHyEn6N0hAqUBp' };
+      const kidsDictionaryData = { ...mockDictionaryData, categories: [kidsCategoryData] };
+      const kidsCategory = getKidsCategory(kidsDictionaryData);
+      expect(kidsCategory).toBeDefined();
+      expect(kidsCategory?.id).toEqual(kidsCategoryData.id);
+    });
+
+    it('should return undefined if the kids category is not present', () => {
+      const kidsCategory = getKidsCategory(mockDictionaryData);
+      expect(kidsCategory).toBeUndefined();
+    });
+  });
+
+  describe('getPhraseById', () => {
+    it('should return a Phrase object with the given id', () => {
+      const phrase = getPhraseById(mockDictionaryData, 'testPhrase');
+      expect(phrase.getId()).toEqual(mockPhraseData.id);
+    });
+  });
+
+  describe('Phrase', () => {
+    const phrase = new Phrase(mockPhraseData);
+
+    it('should return correct translations', () => {
+      expect(phrase.getTranslation('uk')).toEqual(mockPhraseData.source.translation);
+      expect(phrase.getTranslation(mockCountryVariantCZ)).toEqual(mockPhraseData.main.translation);
+    });
+
+    it('should return correct transcriptions', () => {
+      expect(phrase.getTranscription('uk')).toEqual(mockPhraseData.source.transcription);
+      expect(phrase.getTranscription(mockCountryVariantCZ)).toEqual(mockPhraseData.main.transcription);
+    });
+
+    it('should return correct sound URLs', () => {
+      expect(phrase.getSoundUrl('uk')).toEqual(mockPhraseData.source.sound_url);
+      expect(phrase.getSoundUrl(mockCountryVariantCZ)).toEqual(mockPhraseData.main.sound_url);
+    });
+
+    it('should return the correct image URL', () => {
+      expect(phrase.getImageUrl()).toEqual(mockPhraseData.image_url);
+    });
+
+    it('should return the correct ID', () => {
+      expect(phrase.getId()).toEqual(mockPhraseData.id);
+    });
+  });
+
+  // Test fetching functions
+  describe('fetching functions', () => {
+    const mockFetch = fetch as jest.MockedFunction<typeof fetch>;
+
+    it('should fetch raw dictionary data', async () => {
```

Comment:
I don't understand what this test case is testing. The only thing fetchRawDictionary does is construct the correct dictionary URL and you are not testing that here.

Reply:
First fetch returns DictionaryData and second with filtered category. So test is just mocking fetch and return expected object. We can remove if does not make sense